tests/MongoDB.Driver.Tests/MongoClientTests.cs

```diff
@@ -206,6 +210,33 @@ public void UsesSameClusterWhenReadPreferenceTagsAreTheSame()
             databaseNames.Should().Equal(operationResult["databases"].AsBsonArray.Select(record => record["name"].AsString));
         }
 
+        // will probably need to move this test?
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void ReadPreference_should_not_be_sent_to_standalone_server(
+            [Values(false, true)] bool async)
+        {
+            RequireServer.Check().ClusterType(ClusterType.Standalone);
+            var eventCapturer = new EventCapturer().Capture<CommandStartedEvent>(e => e.CommandName.Equals("find"));
+            using (var subject = new DisposableMongoClient(CreateDisposableClient(eventCapturer).WithReadPreference(ReadPreference.Secondary)))
+            {
+                var database = subject.GetDatabase(DriverTestConfiguration.DatabaseNamespace.DatabaseName);
+                var collection = database.GetCollection<BsonDocument>(DriverTestConfiguration.CollectionNamespace.CollectionName);
+
+                if (async)
+                {
+                    collection.FindAsync("{x: 2}").GetAwaiter().GetResult();
+                }
+                else
+                {
+                    collection.FindSync("{x: 2}");
+                }
+            }
+
+            var result = eventCapturer.Events[0].ToBsonDocument();
```

Comment:
Also, since you are capturing only `CommandStartedEvent` now, It looks like it's enough just to convert the event to `CommandStartedEvent`.

Reply:
I'm not sure what you mean, let me know if it still needs fixing after the most recent change.

Reply:
I meant code like this:

            var resultCommand = ((CommandStartedEvent)eventCapturer.Events[0]).Command;
            resultCommand.Contains("$readPreference").Should().BeTrue();

NOTE: this test has been passed on `replica set` too :)

Reply:
I took your code suggestion for the most recent commit, and it seems like it fails on replica set now.